core/src/components/tab-button/tab-button.md.scss

```diff
@@ -75,6 +75,30 @@
   height: $tab-button-md-badge-size-empty;
 }
 
+:host ::slotted(ion-badge[vertical]) {
+  --badge-padding-top: 5px;
```

Comment:
Remove this new CSS variable `--badge-*` since they're not needed to implement the variables. It should follow the structure of the other Sass variables which are hard coded.